src/main/mondrian/rolap/agg/SegmentCacheManager.java

```diff
@@ -307,6 +308,94 @@ public SegmentCacheManager(MondrianServer server) {
         }
 
         compositeCache = new CompositeSegmentCache(segmentCacheWorkers);
+
+        // syncing elements that already exist in external cache
+        // is not possible now - have to wait until the schema has been loaded
+        // thus, let's merely store headers
+        // in-memory cache will not be double-indexed,
+        // because at this point it is empty
+        starFactTablesToSync = pickStarTablesFrom(
```

Comment:
This won't work. There might be new cache segments by the time this is needed. Plus it's a bad practice to build-in such stateful variables.